conan/api/subapi/list.py

```diff
@@ -127,29 +129,31 @@ def select(self, pattern, package_query=None, remote=None):
                 prefs = []
                 if "*" not in pattern.package_id and pattern.prev is not None:
                     prefs.append(PkgReference(rrev, package_id=pattern.package_id))
+                    packages = {}
                 else:
-                    packages = self.conan_api.list.packages_configurations(rrev, remote)
+                    packages = self.packages_configurations(rrev, remote)
                     if package_query is not None:
-                        packages = self.conan_api.list.filter_packages_configurations(packages,
-                                                                                      package_query)
+                        packages = self.filter_packages_configurations(packages, package_query)
```

Comment:
might it be sub-optimal? as `self.packages_configurations` may generate a list of billions items, and then apply a filter over it just to shrink it to the couple of items?

Reply:
There are no packages with a billion binaries. Largest use case will be ConanCenter with +100.
In any case, the server is not capable of filtering packages, so it should be done this way anyway.